internal/connector/connector.go

```diff
@@ -471,7 +474,10 @@ func syncGrantsToDestination(
 ) error {
 	var latestIndex int64 = 1
 
-	sync := func() error {
+	sync := func(ctx context.Context) error {
+		ctx, cancel := context.WithTimeout(ctx, 7*time.Minute)
```

Comment:
I made this 7 minutes, just in case kube API requests are slower than we expect and it takes a while to apply them.